omnipaxos/src/omni_paxos.rs

```diff
@@ -277,37 +391,17 @@ where
     }
 }
 
-/// Used for proposing reconfiguration of the cluster.
-#[derive(Debug, Clone)]
-pub struct ReconfigurationRequest {
-    /// The id of the servers in the new configuration.
-    pub(crate) new_configuration: Vec<NodeId>,
-    /// Optional metadata to be decided with the reconfiguration.
-    pub(crate) metadata: Option<Vec<u8>>,
-}
-
-impl ReconfigurationRequest {
-    /// create a `ReconfigurationRequest`.
-    /// # Arguments
-    /// * `new_configuration`: The pids of the nodes in the new configuration.
-    /// * `metadata`: Some optional metadata in raw bytes. This could include some auxiliary data for the new configuration to start with.
-    pub fn with(new_configuration: Vec<NodeId>, metadata: Option<Vec<u8>>) -> Self {
-        Self {
-            new_configuration,
-            metadata,
-        }
-    }
-}
-
-/// An error returning the proposal that was failed due to that the current configuration is stopped.
+/// An error indicating a failed proposal due to the current cluster configuration being already stopped
+/// or due to an invalid proposed configuration. Returns the failed proposal.
 #[allow(missing_docs)]
```

Comment:
Let's describe each variant since they are not very intuitive i.e., describe that those PendingReconfig variants are due to a pending reconfiguration and they return the failed entry.
```suggestion
```